### PR TITLE
Export analysis files

### PR DESCRIPTION
**Pull request by as2388**
Requires https://github.com/AfricasVoices/Test-Pipeline-Engagement-DB/pull/78 to test fully.

### CODE REVIEW

File: src/engagement_db_to_analysis/analysis_files.py [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+
+def _get_analysis_file_headers(column_configs):
+    """
+    Gets the headers for an analysis file.
+
+    The headers are:
+     - "participant_uuid"
+     - Labels for each normal code scheme, in matrix format e.g. "age:25", "s01e01:healthcare".
+     - Raw messages for each dataset.
+
```
Comment (IsaackMwenda) [inline src/engagement_db_to_analysis/analysis_files.py:37]: - Optional special group membership e.g listening_group_member

File: src/engagement_db_to_analysis/analysis_files.py [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+    The production file contains the participant uuid and all the raw_datasets only.
+
+    :param traced_data_iterable: Data to export.
+    :type traced_data_iterable: iterable of core_data_modules.traced_data.TracedData
+    :param analysis_config: Configuration for the export.
+    :type analysis_config: src.engagement_db_to_analysis.configuration.AnalysisConfiguration
+    :param export_path: Path to export the file to.
+    :type export_path: str
+    """
```
Comment (IsaackMwenda) [inline src/engagement_db_to_analysis/analysis_files.py:22]:
```suggestion
    """
        log.info("Exporting production CSV...")

```

**Review by IsaackMwenda — APPROVED** [review]

File: src/engagement_db_to_analysis/analysis_files.py (hunk 1 above)
Comment (as2388) [reply src/engagement_db_to_analysis/analysis_files.py:37]: Will add this to the docstring when we add the behaviour

File: src/engagement_db_to_analysis/analysis_files.py (hunk 2 above)
Comment (as2388) [reply src/engagement_db_to_analysis/analysis_files.py:22]: Thanks, added log.info(f"Exporting production file to '{export_path}'...")

**Review by DanielMwendwa — APPROVED** [review]

File: src/engagement_db_to_analysis/engagement_db_to_analysis.py [hunk 3]
```diff
@@ -162,10 +150,11 @@ def generate_analysis_files(user, pipeline_config, engagement_db, cache_path=Non
     impute_codes_by_column_traced_data(user, participants_by_column, pipeline_config.analysis_configs.dataset_configurations)
 
     # Export to hard-coded files for now.
-    # TODO: Only export a production file for messages (exporting both for now to aid with debugging)
     # TODO: Export to a directory passed in on the command line rather than a hard-coded analysis folder.
-    export_production_file(messages_by_column, pipeline_config.analysis_configs, "analysis/messages-production.csv")
-    export_production_file(participants_by_column, pipeline_config.analysis_configs, "analysis/participants-production.csv")
+    export_production_file(messages_by_column, pipeline_config.analysis_configs, "analysis/production.csv")
+
+    export_analysis_file(messages_by_column, pipeline_config.analysis_configs.dataset_configurations, "analysis/messages.csv")
```
[the remark this answers is not in the document]
Comment (as2388) [reply src/engagement_db_to_analysis/engagement_db_to_analysis.py:152]: Thanks, I' have converted this to a todo in the code as all these functions are inconsistent in which they accept so we can open a new PR that cleans them up in future

**Pull request merged**